protocol/protocol.tex

```diff
@@ -7643,92 +9143,179 @@
 \end{formulae}
+} %nufive
 
 \begin{securityrequirements}
-  \item $\HomomorphicPedersenCommitAlg$ must be a computationally binding and at least
-        computationally hiding \commitmentScheme, for a given personalization input $D$.
-  \item $\ValueCommitAlg$ must be a computationally binding and at least
-        computationally hiding \commitmentScheme.
+  \item $\HomomorphicPedersenCommitAlg{Sapling}$\nufive{ and $\HomomorphicPedersenCommitAlg{Orchard}$}
+        must be\notnufive{ a} computationally binding and at least computationally hiding
+        \notnufive{\commitmentScheme}\notbeforenufive{\commitmentSchemes}, for a given personalization input $D$.
+  \item $\ValueCommitAlg{Sapling}$\nufive{ and $\ValueCommitAlg{Orchard}$} must be\notnufive{ a} computationally
+        binding and at least computationally hiding \notnufive{\commitmentScheme}\notbeforenufive{\commitmentSchemes}.
 \end{securityrequirements}
 
 (They are in fact unconditionally hiding \commitmentSchemes.)
-}
+} %sapling
+
+
+\nufive{
+\introsection
+\extralabel{concreteorchardnotecommit}{\lsubsubsubsection{Sinsemilla commitments}{concretesinsemillacommit}}
+
+Let $\ExtractP$ be as defined in \crossref{concreteextractorpallas}.
+
+\crossref{concretesinsemillahash} defines a \xSinsemillaHash construction.
+We construct \defining{\xSinsemillaCommitments} by reusing that construction,
+and adding a randomized point on the \pallasCurve (see \crossref{pallasandvesta}):
+
+\begin{formulae}
+  \item $\SinsemillaCommit{r}(D, M) :=
+           \SinsemillaHashToPoint(D \bconcat \ascii{-M}, M) + \scalarmult{r}{\GroupPHash\Of{D \bconcat \ascii{-r}, \ascii{}}}$
+  \item $\SinsemillaShortCommit{r}(D, M) :=
+           \ExtractP\Of{\SinsemillaCommit{r}(D, M)}$
+\end{formulae}
+
+See \todo{...} for rationale and efficient circuit implementation of this function.
+
+The commitment scheme $\NoteCommitAlg{Orchard}$ specified in \crossref{abstractcommit} is
+instantiated as follows using $\SinsemillaCommitAlg$:
+
+\begin{formulae}
+  \item $\NoteCommit{Orchard}{\NoteCommitRand}(\DiversifiedTransmitBaseRepr, \DiversifiedTransmitPublicRepr, \Value) :=
+           \SinsemillaCommit{\NoteCommitRand}\!\big(\ascii{z.cash:Orchard-NoteCommit},
+             \DiversifiedTransmitBaseRepr \bconcat \DiversifiedTransmitPublicRepr \bconcat \ItoLEBSP{64}(\Value)\kern-0.15em\big)$
+  \item $\NoteCommitGenTrapdoor{Orchard}()$ generates the uniform distribution on $\GF{\ParamP{r}}$.
+\end{formulae}
+
+The commitment scheme $\CommitIvkAlg$ specified in \crossref{abstractcommit} is
+instantiated as follows using $\SinsemillaCommitAlg$:
+
+\begin{formulae}
+  \item $\CommitIvk{\CommitIvkRand}(\AuthSignPublic, \NullifierKey) :=
+           \SinsemillaShortCommit{\NoteCommitRand}\left(\ascii{z.cash:Orchard-CommitIvk},
+             \ItoLEBSP{\ScalarLength{Orchard}}(\AuthSignPublicRepr) \bconcat \ItoLEBSP{\ScalarLength{Orchard}}\NullifierKeyRepr\right) \mod{\ParamP{r}}$
```

Comment:
Separately, I think it's a type error to pass a bitstring into `I2LEBSP`:
```suggestion
             \ItoLEBSP{\ScalarLength{Orchard}}(\AuthSignPublic) \bconcat \ItoLEBSP{\ScalarLength{Orchard}}\NullifierKey\right) \mod{\ParamP{r}}$
```
I presume that the intention is we have both `ak` and `nk` represented as (base field) integers by the time we call `I2LEBSP`.

Reply:
Yes this is a type error.